File: 874-backspace-string-compare/solution.java
class Solution {
    public boolean backspaceCompare(String S, String T) {
        if (g(S).equals(g(T))){
            return true;
        }
        else{
            return false;
        }

    }
    public static Stack<Character> g(String h){
        Stack<Character> x = new Stack<>();
        for (int i = 0; i < h.length(); i++) {
            if (h.charAt(i) == '#' && x.size() > 0){
                x.pop();
            }
            else if (h.charAt(i) != '#'){
                x.push(h.charAt(i));
            }
        }
        return x;
    }
}

File: 874-backspace-string-compare/solution.go
func backspaceCompare(s string, t string) bool {
	for i := 0; i < len(s); i++ {
		if i > 0 && s[i] == '#' {
			s = s[:i-1] + s[i+1:]
			i -= 2
		} else if s[i] == '#' {
			s = s[1:]
			i -= 1
		}
	}
	for i := 0; i < len(t); i++ {
		if i > 0 && t[i] == '#' {
			t = t[:i-1] + t[i+1:]
			i -= 2
		} else if t[i] == '#' {
			t = t[1:]
			i -= 1
		}
	}
	return s == t
}